File: MouseController/MouseController.m
//
//  MouseController.m
//  testMouseCursor
//
//  Created by James Tang on 30/9/2019.
//  Copyright © 2019 James Tang. All rights reserved.
//

#import "MouseController.h"
@import AppKit;

@implementation MouseController

- (void)changeCursor:(Cursor)cursor {
    switch (cursor) {
        case CursorText:
            [NSCursor.IBeamCursor set];
            break;
        case CursorArrow:
            [NSCursor.arrowCursor set];
            break;
    }
}

- (void)setCursorImageData:(NSData *)image hotSpot:(CGPoint)point {
    NSImage *cursorImage = [[NSImage alloc] initWithData: image];
    NSCursor *cursor = [[NSCursor alloc] initWithImage:cursorImage hotSpot:point];
    [cursor set];
}

@end

File: MouseController/MouseController.swift
//
//  MouseController.swift
//  MouseController
//
//  Created by James Tang on 29/9/2019.
//  Copyright © 2019 James Tang. All rights reserved.
//

import AppKit

public class MouseController: NSObject, MouseProtocol {

    var notificationHandler: Any!

    override required public init() {
        super.init()
        let notificationHandler = NotificationCenter.default.addObserver(forName: NSNotification.Name("ChangeCursorNotification"), object: nil, queue: OperationQueue.main, using: { [weak self] (notification) in
            self?.handleCursorNotification(notification: notification)
        })
        self.notificationHandler = notificationHandler
    }

    func handleCursorNotification(notification: Notification) {
        Swift.print("handleCursorNotification \(notification)")
    }

    public func changeCursor(_ cursor: Cursor) {
        switch cursor {
        case .text:
            NSCursor.iBeam.set()
        case .arrow:
            NSCursor.arrow.set()
        }
    }

}
